cpp/src/slice2py/PythonUtil.cpp

```diff
@@ -1776,14 +1763,8 @@ Slice::Python::CodeVisitor::writeAssign(const MemberInfo& info)
     StructPtr st = dynamic_pointer_cast<Struct>(info.dataMember->type());
     if (st && !info.dataMember->optional())
     {
-        _out << nl << "if " << paramName << " is Ice._struct_marker:";
-        _out.inc();
-        _out << nl << "self." << memberName << " = " << getSymbol(st) << "()";
-        _out.dec();
-        _out << nl << "else:";
-        _out.inc();
-        _out << nl << "self." << memberName << " = " << paramName;
-        _out.dec();
+        _out << nl << "self." << memberName << " = " << paramName << "if " << paramName << " is not None else "
+             << getSymbol(st) << "()";
```

Comment:
I replaced `_struct_marker` with `None` it is not clear why we need to have default values for all parameters. I need to review the generated code for constructors.